syntax = "proto3";
import "nanopb.proto";

option go_package = "pb";

message CentralMessage {
    sint32 rssi = 1;
    bytes RemoteId = 2 [(nanopb).max_size = 6, (nanopb).fixed_length = true];
}

message SensorMessage {
    repeated Reading Readings = 1 [(nanopb).max_count = 5];
    CentralMessage ProxyCentral = 2 [(nanopb).proto3 = false];
    sint32 rssi = 3;
    int32 BatteryVoltage = 4;
}

message Reading {
    bytes Id = 1 [(nanopb).max_size = 6];
    sint32 Temperature = 2;
    int32 Humidity = 3;
    int32 Battery = 4;
    float Current = 5;
}
